### ISSUE TITLE
lsp--text-document-hover-string symbol detection

### ISSUE THREAD

**Issue by alanz** [emacs-lsp/lsp-mode#192]
At the moment, the decision to request the hover string is based on a call to `symbol-at-point`.

Haskell has source like

```haskell
  let uri = vtdi ^. J.uri
```
where the `^.` is a valid symbol. 

Placing the cursor on this does not generate a hover request.

**Comment by vibhavp** [comment]
Yeah, this is a syntax issue. Does `haskell-mode` have ways to get the current symbol under point?

**Comment by sebastiencs** [comment]
From the documentation of `thing-at-point` it's possible to define what is a symbol.

There are 3 functions to define: `forward-op` `beginning-op` and `end-op` for `symbol`.
You can find more information in the file [thingatpt.el](https://github.com/emacs-mirror/emacs/blob/master/lisp/thingatpt.el)

Not sure if it's the easiest way to solve your issue, but it can be a start

**Comment by alanz** [comment]
Thanks. And the documentation implies that those things should be set according to the major mode. So perhaps `haskell-mode` needs to be adjusted.

**Comment by alanz** [comment]
I can confirm that I can get hover info for symbols for haskell-lsp, so agree with this being closed. Thanks.
